src/Raven.Client/Http/NodeSelector.cs

```diff
@@ -28,6 +29,43 @@ public NodeSelectorState(Topology topology)
                 UnlikelyEveryoneFaultedChoiceIndex = 0;
             }
 
+            public NodeSelectorState(Topology topology, NodeSelectorState prevState)
+            {
+                Topology = topology;
+                Nodes = topology.Nodes;
+                Failures = new int[topology.Nodes.Count];
+                FastestRecords = new int[topology.Nodes.Count];
+                UnlikelyEveryoneFaultedChoiceIndex = 0;
+
+                var fastestNode = prevState.Nodes.ElementAt(Fastest);
```

Comment:
`Fastest` is always a part of `prevState.Nodes` and derived from it so I don't see how this could happen. 'Nodes[Fastest]` is used in to fetch the fastest in this fashion throughout the code